src/components/utils/component-resolver.tsx

```diff
@@ -1,27 +1,97 @@
+'use client';
+
 import React, { useMemo } from 'react';
+import { useContentfulContext } from '@/contentful-context';
+import { componentGqlMap, componentMap } from '@/mapping';
 
+let previousComponent: string | null = null;
 interface Props {
-  componentProps: {
-    sys: { id: string };
-    __typename: string;
-    [k: string]: any;
-  };
-
-  /**
-   * forces to do a graqhql request to get its content, instead
-   * of expecting content is provided trough `props.componentProps`:
-   */
-  forceGql?: boolean;
-
-  className?: string;
-  inline?: boolean;
+    componentProps: {
+        sys: { id: string };
+        __typename: string;
+    };
+
+    /**
+     * forces to do a graqhql request to get its content, instead
+     * of expecting content is provided trough `props.componentProps`:
+     */
+    forceGql?: boolean;
+
+    className?: string;
+    inline?: boolean;
 }
 
-export const ComponentResolver = () => {
-  return (
-    <div>
-      ComponentResolver TODO: see
-      https://github.com/contentful/template-marketing-webapp-nextjs
-    </div>
-  );
+
+interface index {
+    [key: string]: any
+}
+
+export const ComponentResolver = (props: Props) => {
+    const { componentProps, inline = false } = props;
+    const { previewActive } = useContentfulContext();
+
+    const { locale } = useContentfulContext();
+
+    // @ts-ignore
+    const ComponentGql = componentGqlMap[componentProps.__typename];
```

Comment:
why is this ts ignore here?

Reply:
removing the @ts-ignore results in errors on the typings.
const ComponentGql = (componentGqlMap as {type})[componentProps.__typename];
Replace {type} with the proper typing on componentGqlMap. I've seen this module is lazy loaded, so best to separate the typings so you can still use them elsewhere.

Reply:
No idea why but for some reason the error is gone